generator/src/FluentStageFactoryGenerator.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\CodeGenerator;
+
+use MongoDB\Builder\Pipeline;
+use MongoDB\Builder\Stage;
+use MongoDB\Builder\Type\StageInterface;
+use MongoDB\CodeGenerator\Definition\GeneratorDefinition;
+use Nette\PhpGenerator\ClassType;
+use Nette\PhpGenerator\Method;
+use Nette\PhpGenerator\Parameter;
+use Nette\PhpGenerator\PhpFile;
+use Nette\PhpGenerator\PhpNamespace;
+use Nette\PhpGenerator\TraitType;
+use ReflectionClass;
+
+use function array_key_last;
+use function array_map;
+use function assert;
+use function file_get_contents;
+use function implode;
+use function sprintf;
+
+/**
+ * Generates a fluent factory class for aggregation pipeline stages.
+ * The method definition is based on all the public static methods
+ * of the Stage class.
+ */
+class FluentStageFactoryGenerator extends OperatorGenerator
+{
+    private const FACTORY_CLASS = Stage::class;
+
+    public function generate(GeneratorDefinition $definition): void
+    {
+        $this->writeFile($this->createFluentFactoryClass($definition));
+    }
+
+    private function createFluentFactoryClass(GeneratorDefinition $definition): PhpNamespace
+    {
+        $namespace = new PhpNamespace($definition->namespace);
+        $class = $namespace->addClass('FluentFactory');
+
+        $namespace->addUse(self::FACTORY_CLASS);
+        $namespace->addUse(StageInterface::class);
+        $namespace->addUse(Pipeline::class);
+
+        $class->addProperty('pipeline')
+            ->setType('array')
+            ->setComment('@var list<StageInterface>')
+            ->setValue([]);
+        $class->addMethod('getPipeline')
+            ->setReturnType(Pipeline::class)
+            ->setBody(<<<'PHP'
+                return new Pipeline(...$this->pipeline);
+                PHP);
+
+        $this->addUsesFrom(self::FACTORY_CLASS, $namespace);
+        $staticFactory = ClassType::from(self::FACTORY_CLASS);
+        assert($staticFactory instanceof ClassType);
+
+        // Import the methods customized in the factory class
+        foreach ($staticFactory->getMethods() as $method) {
+            if (! $method->isPublic()) {
+                continue;
+            }
+
```

Comment:
This can be removed now that the check is in `addMethod`:
```suggestion
```

Reply:
Oups.